### ISSUE TITLE
Improving Array2d API

### ISSUE THREAD

**Issue by Ducasse** [guillep/pharo-core#11997]
```
Array2DTest >> testRowsColumnsPut
	| m |
	m := Array2D rows: 3 columns: 2.
	m rowsColumnsPut: [ :row :column | column * 10 + row ].
	self assert: (m at: 1 at: 1) equals: 11.
	self assert: (m at: 1 at: 2) equals: 21.
	self assert: (m at: 2 at: 1) equals: 12.
	self assert: (m at: 2 at: 2) equals: 22.
	self assert: (m at: 3 at: 1) equals: 13.
	self assert: (m at: 3 at: 2) equals: 23.
```

**Comment by Ducasse** [comment]
```
Array2D class >> rows: rowNumber columns: columnNumber tabulate: aTwoArgumentBlock
	"Answer a new array2d of the given dimensions where
	 result at: i at: j is aTwoArgumentBlock value: i value: j"

	| newArray |
	newArray := self rows: rowNumber columns: columnNumber.
	newArray rowsColumnsPut: aTwoArgumentBlock.
	^ newArray
```

**Comment by Ducasse** [comment]
```
Array2D >> rowsColumnsPut: aTwoArgumentBlock
	"Set the value at row,colum as the value of aTwoArgumentBlock with row and column as inputs."

	| i |
	i := 0.
	1 to: numberOfRows do: [:row |
		1 to: numberOfColumns do: [:column |
			contents at: (i := i + 1) put: (aTwoArgumentBlock value: row value: column)]].
```

**Comment by Ducasse** [comment]
This instance side method is important because sometimes we need be able to do an action in the block requiring that the array2D is already created. So we should make sure that the tabulate: can be done after the creation. 
Previously this logic was in the class creation method and it could not be reused on the instance side.

**Comment by Ducasse** [comment]
In addition we can have

```
rowsColumnsDo: aTwoArgumentBlock
	"Perform aTwoArgumentBlock with row and column as inputs"

	| i |
	i := 0.
	1 to: numberOfRows do: [:row |
		1 to: numberOfColumns do: [:column |
			aTwoArgumentBlock value: row value: column]].
```

**Comment by Inao0** [comment]
This issue was fixed in the pharo-containers depot.
The fix of the pull request #12130 was merged in https://github.com/pharo-containers/Containers-Array2D/pull/15.
I think this issue should be closed.
